Subagent tool naming validation error when using Anthropic provider with OAuth

## Description

When attempting to launch the `typescriptexpert` subagent (configured with the `claudecode` profile using Anthropic provider with OAuth), the following error occurs:

```
400 {"type":"error","error":{"type":"invalid_request_error","message":"tools.17.custom.name: String should match pattern '^[a-zA-Z0-9_-]{1,128}$'"},"request_id":"req_011CUsmyGA9EBGJqSFtdNwrw"}
```

## Steps to Reproduce

1. Configure a subagent with an Anthropic provider profile using OAuth
2. Attempt to launch the subagent using the `task` tool
3. Observe the error about tool name validation

## Expected Behavior

The subagent should launch successfully without tool naming validation errors.

## Actual Behavior

The API returns a 400 error indicating that tool #17 has a custom name that doesn't match the required pattern `^[a-zA-Z0-9_-]{1,128}$`.

## Environment

- Provider: Anthropic (OAuth)
- Profile: claudecode
- Subagent: typescriptexpert

## Additional Context

This error suggests that one of the tools being sent to the Anthropic API has an invalid name, possibly containing special characters or spaces that violate the naming constraints.

## Identified Problematic Tool

After searching through the codebase, I've confirmed that there is **exactly one tool** with a name that violates Anthropic's naming pattern:

### The Culprit: `self.emitvalue`

**Location:** `packages/core/src/core/subagent.ts`, line 1666

```typescript
private getScopeLocalFuncDefs() {
  const emitValueTool: FunctionDeclaration = {
    name: 'self.emitvalue',  // [ERROR] Contains a dot - INVALID for Anthropic
    description: `* This tool emits A SINGLE return value from this execution...`,
    parameters: {
      // ... parameters definition
    },
  };

  return [emitValueTool];
}
```

**Why it causes the error:**
- This is a scope-local tool that gets added to the function declarations when running a subagent
- The dot character (`.`) in `self.emitvalue` violates Anthropic's tool naming requirement: `^[a-zA-Z0-9_-]{1,128}$`
- This tool is only added when running subagents, which is why the error only occurs with subagent executions

### All Other Core Tools Are Valid

I verified all other core tools in the codebase, and they all use valid names:

**Valid tool names (from `packages/core/src/tools/`):**
- `run_shell_command`
- `glob`
- `list_directory`
- `list_subagents`
- `replace`
- `search_file_content`
- `task`
- `todo_pause`
- `todo_read`
- `todo_write`
- `delete_line_range`
- `google_web_search`
- `insert_at_line`
- `read_file`
- `read_line_range`
- `read_many_files`
- `web_fetch`
- `write_file`
- `save_memory`

All of these names only use alphanumeric characters and underscores, so they comply with Anthropic's requirements.

### Impact

This issue:
- **Only affects subagent execution** when using Anthropic provider (Claude models)
- **Does not affect regular tool calls** or other providers
- **Blocks all subagent usage** with Anthropic/Claude, making the `task` tool unusable with those models

### Recommended Fix

The simplest fix is to rename `self.emitvalue` to `self_emitvalue` in `subagent.ts` line 1666:

```typescript
const emitValueTool: FunctionDeclaration = {
  name: 'self_emitvalue',  // [OK] Valid for all providers
  description: `* This tool emits A SINGLE return value from this execution...`,
  // ... rest unchanged
};
```

This is a minimal change that:
- [OK] Fixes the Anthropic validation error
- [OK] Maintains backward compatibility (tool is internal to subagents)
- [OK] Doesn't require any sanitization logic
- [OK] Uses a more standard naming convention (snake_case)

However, if we want to be defensive and future-proof, we should **also** add the sanitization logic proposed in my previous comment to catch any future issues with tool names from MCP servers or discovered tools.

--- a/packages/cli/src/config/config.ts
+++ b/packages/cli/src/config/config.ts
@@ -75,7 +75,7 @@ export const READ_ONLY_TOOL_NAMES = [
   'web_fetch',
   'todo_read',
   'task',
-  'self.emitvalue',
+  'self_emitvalue',
 ] as const;
 
 const EDIT_TOOL_NAME = 'replace';

--- a/packages/core/src/core/subagent.test.ts
+++ b/packages/core/src/core/subagent.test.ts
@@ -967,7 +967,7 @@ describe('subagent.ts', () => {
           'you MUST emit the required output variables',
         );
         expect(systemInstruction).toContain(
-          "Use 'self.emitvalue' to emit the 'result1' key",
+          "Use 'self_emitvalue' to emit the 'result1' key",
         );
       });
 
@@ -1151,7 +1151,7 @@ describe('subagent.ts', () => {
         expect(scope.output.terminate_reason).toBe(SubagentTerminateMode.GOAL);
       });
 
-      it('should handle self.emitvalue and terminate with GOAL when outputs are met', async () => {
+      it('should handle self_emitvalue and terminate with GOAL when outputs are met', async () => {
         const { config } = await createMockConfig();
         const outputConfig: OutputConfig = {
           outputs: { result: 'The final result' },
@@ -1163,7 +1163,7 @@ describe('subagent.ts', () => {
           createMockStream([
             [
               {
-                name: 'self.emitvalue',
+                name: 'self_emitvalue',
                 args: {
                   emit_variable_name: 'result',
                   emit_variable_value: 'Success!',
@@ -1482,7 +1482,7 @@ describe('subagent.ts', () => {
             'stop',
             [
               {
-                name: 'self.emitvalue',
+                name: 'self_emitvalue',
                 args: {
                   emit_variable_name: 'required_var',
                   emit_variable_value: 'Here it is',
@@ -1536,20 +1536,20 @@ describe('subagent.ts', () => {
           createMockStream([
             [
               {
-                name: 'self.emitvalue',
+                name: 'self_emitvalue',
                 args: { emit_variable_name: 'loop', emit_variable_value: 'v1' },
               },
             ],
             [
               {
-                name: 'self.emitvalue',
+                name: 'self_emitvalue',
                 args: { emit_variable_name: 'loop', emit_variable_value: 'v2' },
               },
             ],
             // This turn should not happen
             [
               {
-                name: 'self.emitvalue',
+                name: 'self_emitvalue',
                 args: { emit_variable_name: 'loop', emit_variable_value: 'v3' },
               },
             ],

--- a/packages/core/src/core/subagent.ts
+++ b/packages/core/src/core/subagent.ts
@@ -701,7 +701,7 @@ export class SubAgentScope {
           const schedulerRequests: ToolCallRequestInfo[] = [];
 
           for (const request of toolRequests) {
-            if (request.name === 'self.emitvalue') {
+            if (request.name === 'self_emitvalue') {
               manualParts.push(...this.handleEmitValueCall(request));
             } else {
               schedulerRequests.push(request);
@@ -795,7 +795,7 @@ export class SubAgentScope {
 
         const nudgeMessage = `You have stopped calling tools but have not emitted the following required variables: ${remainingVars.join(
           ', ',
-        )}. Please use the 'self.emitvalue' tool to emit them now, or continue working if necessary.`;
+        )}. Please use the 'self_emitvalue' tool to emit them now, or continue working if necessary.`;
 
         this.logger.debug(
           () =>
@@ -1057,7 +1057,7 @@ export class SubAgentScope {
 
           const nudgeMessage = `You have stopped calling tools but have not emitted the following required variables: ${remainingVars.join(
             ', ',
-          )}. Please use the 'self.emitvalue' tool to emit them now, or continue working if necessary.`;
+          )}. Please use the 'self_emitvalue' tool to emit them now, or continue working if necessary.`;
 
           this.logger.debug(
             () =>
@@ -1107,7 +1107,7 @@ export class SubAgentScope {
   /**
    * Processes a list of function calls, executing each one and collecting their responses.
    * This method iterates through the provided function calls, executes them using the
-   * `executeToolCall` function (or handles `self.emitvalue` internally), and aggregates
+   * `executeToolCall` function (or handles `self_emitvalue` internally), and aggregates
    * their results. It also manages error reporting for failed tool executions.
    * @param {FunctionCall[]} functionCalls - An array of `FunctionCall` objects to process.
    * @param {ToolRegistry} toolRegistry - The tool registry to look up and execute tools.
@@ -1141,7 +1141,7 @@ export class SubAgentScope {
       let toolResponse;
 
       // Handle scope-local tools first.
-      if (functionCall.name === 'self.emitvalue') {
+      if (functionCall.name === 'self_emitvalue') {
         const valName = String(requestInfo.args['emit_variable_name']);
         const valVal = String(requestInfo.args['emit_variable_value']);
         this.output.emitted_vars[valName] = valVal;
@@ -1388,7 +1388,7 @@ export class SubAgentScope {
     }
 
     const errorMessage =
-      'self.emitvalue requires emit_variable_name and emit_variable_value arguments.';
+      'self_emitvalue requires emit_variable_name and emit_variable_value arguments.';
     this.logger.warn(
       () => `Subagent ${this.subagentId} failed to emit value: ${errorMessage}`,
     );
@@ -1658,12 +1658,12 @@ export class SubAgentScope {
 
   /**
    * Returns an array of FunctionDeclaration objects for tools that are local to the subagent's scope.
-   * Currently, this includes the `self.emitvalue` tool for emitting variables.
+   * Currently, this includes the `self_emitvalue` tool for emitting variables.
    * @returns An array of `FunctionDeclaration` objects.
    */
   private getScopeLocalFuncDefs() {
     const emitValueTool: FunctionDeclaration = {
-      name: 'self.emitvalue',
+      name: 'self_emitvalue',
       description: `* This tool emits A SINGLE return value from this execution, such that it can be collected and presented to the calling function.
         * You can only emit ONE VALUE each time you call this tool. You are expected to call this tool MULTIPLE TIMES if you have MULTIPLE OUTPUTS.`,
       parameters: {
@@ -1746,10 +1746,10 @@ export class SubAgentScope {
     // Add instructions for emitting variables if needed.
     if (this.outputConfig && this.outputConfig.outputs) {
       let outputInstructions =
-        '\n\nAfter you have achieved all other goals, you MUST emit the required output variables. For each expected output, make one final call to the `self.emitvalue` tool.';
+        '\n\nAfter you have achieved all other goals, you MUST emit the required output variables. For each expected output, make one final call to the `self_emitvalue` tool.';
 
       for (const [key, value] of Object.entries(this.outputConfig.outputs)) {
-        outputInstructions += `\n* Use 'self.emitvalue' to emit the '${key}' key, with a value described as: '${value}'`;
+        outputInstructions += `\n* Use 'self_emitvalue' to emit the '${key}' key, with a value described as: '${value}'`;
       }
       finalPrompt += outputInstructions;
     }

